native/lambda_game_engine/src/game.rs

```diff
@@ -525,10 +530,18 @@ fn apply_zone_effects(players: &mut HashMap<u64, Player>, zone: &Zone) {
         });
 
         outside_players.into_iter().for_each(|player| {
-            player.apply_effects_if_not_present(
-                &current_modification.outside_radius_effects,
-                EntityOwner::Zone,
-            );
+            if player.status == PlayerStatus::Alive {
+                player.apply_effects_if_not_present(
```

Comment:
I saw you added a `update_status` function to change the status of a player as dead.
However that logic were being used in `decrease_health`. Couldn't we just call to that function to decrease the player HP?

I think updating a player status everywhere could lead to problems in the future where you have to maintain that behavior in every line of code where you touch the HP.
If that's not possible, maybe we can just update its status here, right before the `status == Death` if.

Reply:
Yeah, we noticed that with @manucamejo while doing this, and agreed to leave it as part of a refactor in the near future.